network/crl_test.go

```diff
@@ -0,0 +1,468 @@
+	verifiedChains = append(verifiedChains, []*x509.Certificate{})
+
+	// Read and parse certificates, construct the chain
+	for _, p := range paths {
+		filename := path.Join(prefix, p)
+
+		rawCert, err := ioutil.ReadFile(filename)
+		if err != nil {
+			t.Fatalf("Cannot read certificate: %v\n", err)
+		}
+
+		cert, err := pemToX509Cert(rawCert)
+		if err != nil {
+			t.Fatalf("Cannot parse certificate: %v", err)
+		}
+
+		rawCerts = append(rawCerts, rawCert)
+		verifiedChains[0] = append(verifiedChains[0], cert)
+	}
+
+	// Make sure this function was called with valid arguments
+	for i, cert := range verifiedChains[0] {
+		if i == 0 && cert.IsCA {
+			// Avoid situations where we got two CA certificates,
+			// first one should be used for authentication/encryption
+			t.Fatalf("First certificate in chain (`%s`) should not be CA", cert.Subject.String())
+		}
+
+		var nextCert *x509.Certificate
+		if i < len(verifiedChains[0])-1 {
+			nextCert = verifiedChains[0][i+1]
+		}
+
+		if nextCert != nil {
+			// Every certificate (except last) should be signed by the next one
+			err := cert.CheckSignatureFrom(nextCert)
+			if err != nil {
+				t.Fatalf("Certificate `%s` was not signed by `%s`: %v\n", cert.Subject.String(), nextCert.Subject.String(), err)
+			}
+		}
+	}
+
+	return rawCerts, verifiedChains
+}
+
+func getValidTestChain(t *testing.T) ([][]byte, [][]*x509.Certificate) {
+	return getTestChain(
+		t,
+		TestCertPrefix,
+		TestCertFilename,
+		TestCACertFilename)
+}
+
+// func getValidTestChain3(t *testing.T) ([][]byte, [][]*x509.Certificate) {
+// 	return getTestChain(
+// 		t,
+// 		TestCertPrefix3,
+// 		TestCertFilename3,
+// 		TestICACertFilename3,
+// 		TestCACertFilename3)
+// }
+
+func getInvalidTestChain(t *testing.T) ([][]byte, [][]*x509.Certificate) {
+	return getTestChain(
+		t,
+		TestCertPrefix,
+		TestRevokedCertFilename,
+		TestCACertFilename)
+}
+
+// func getInvalidTestChain3(t *testing.T) ([][]byte, [][]*x509.Certificate) {
+// 	return getTestChain(
+// 		t,
+// 		TestCertPrefix3,
+// 		TestRevokedCertFilename3,
+// 		TestICACertFilename3,
+// 		TestCACertFilename3)
+// }
+
+func TestCRLConfig(t *testing.T) {
+	expectOk := func(uri, fromCert string, cacheSize, cacheTime int) {
+		config, err := NewCRLConfig(uri, fromCert, cacheSize, cacheTime)
+		if config == nil || err != nil {
+			t.Logf("uri=%v, fromCert=%v, cacheSize=%v, cacheTime=%v\n", uri, fromCert, cacheSize, cacheTime)
+			t.Logf("config=%v, err=%v\n", config, err)
+			t.Fatal("Got `nil` result or unexpected error")
+		}
+	}
+
+	expectErr := func(uri, fromCert string, cacheSize, cacheTime int) {
+		config, err := NewCRLConfig(uri, fromCert, cacheSize, cacheTime)
+		if config != nil || err == nil {
+			t.Logf("uri=%v, fromCert=%v, cacheSize=%v, cacheTime=%v\n", uri, fromCert, cacheSize, cacheTime)
+			t.Logf("config=%v, err=%v\n", config, err)
+			t.Fatal("Got unexpected result or `nil` error")
+		}
+	}
+
+	expectOk("", "use", 1, 5)
```

Comment:
will be nice to use constants instead "use" to understand where we can find all acceptable values and which are not correct.

Reply:
Replaced string literals with constants in places like this